src/Raven.Server/ServerWide/ServerStore.cs

```diff
@@ -393,50 +395,77 @@ public void Initialize()
             using (ContextPool.AllocateOperationContext(out TransactionOperationContext ctx))
             {
                 if (TryLoadAuthenticationKeyPairs(ctx) == false)
-                    GenerateAuthenticationKeyPairs(ctx);
+                    GenerateAuthenticationSignetureKeys(ctx);                
             }
+            GenerateAuthenticationBoxKeys();
 
             Task.Run(ClusterMaintenanceSetupTask, ServerShutdown);
         }
 
         public byte[] BoxPublicKey, BoxSecretKey, SignPublicKey, SignSecretKey;
 
-        private void GenerateAuthenticationKeyPairs(TransactionOperationContext ctx)
+        private Tuple<DateTime, string> _clusterToken = Tuple.Create<DateTime, string>(DateTime.MinValue, null);
+
+        internal string GetClusterTokenForNode(JsonOperationContext context)
+        {
+            if (DateTime.UtcNow > _clusterToken.Item1 || _clusterToken.Item2 == null)
```

Comment:
Why wouldn't we generate this on the fly on each call? They are somewhat expensive, but this seems much better to just  generate it each time.
Your code here is dangerous, since we might be giving a token that will expire in 1 ms . 

We aren't going to call this very often, anyway